src/main/java/space/obminyashka/items_exchange/controller/AdvertisementController.java

```diff
@@ -112,6 +115,25 @@ public ResponseEntity<Page<AdvertisementTitleDto>> getPageOfAdvertisementsByKeyw
                 new ResponseEntity<>(allByKeyword, HttpStatus.OK);
     }
 
+    @GetMapping(ApiKey.ADV_SEARCH_PAGINATED_BY_CATEGORY_ID)
+    @Operation(summary = "Find advertisements by category id")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
+            @ApiResponse(responseCode = "404", description = "NOT FOUND")})
+    public Page<AdvertisementTitleDto> getPageOfAdvertisementsByCategoryId(
+            @PathVariable("category_id") Long id,
+            @Parameter(name = "Results page you want to retrieve (0..N). Default value: 0")
+            @RequestParam(value = "page", required = false, defaultValue = "0") @PositiveOrZero int page,
+            @Parameter(name = "Number of records per page. Default value: 12")
+            @RequestParam(value = "size", required = false, defaultValue = "12") @PositiveOrZero int size) throws CategorySizeNotFoundException {
```

Comment:
I used the old CategorySizeNotFoundException, if necessary I can add a new one

Reply:
CategorySize exception describes that size is not available for a selected category. What do you think is that matches your needs?

Reply:
Added new